Jenkinsfile

```diff
@@ -92,56 +92,149 @@ pipeline {
+                  "http://metadata.google.internal/computeMetadata/v1/project/project-id"',
+                  returnStdout: true
+                ).trim()
+                echo "Google Cloud project name: $GCP_PROJECT"
+                unstash 'get_gce_tokens_script'
                 sh '''
-                  retrieve_token() {
-                    local token_format="$1"
-                    local audience="$2"
-
-                    curl \
-                      -s \
-                      -H 'Metadata-Flavor: Google' \
-                      "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=${token_format}&audience=${audience}"
-                  }
-
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/test-app")" > "gcp_token_valid"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-existing")" > "gcp_token_non_existing_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-rooted/test-app")" > "gcp_token_non_rooted_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/test-app")" > "gcp_token_user"
-                  echo "$(retrieve_token "full" "conjur/non-existing/host/test-app")" > "gcp_token_non_existing_account"
-                  echo "$(retrieve_token "full" "invalid_audience")" > "gcp_token_invalid_audience"
-                  echo "$(retrieve_token "standard" "conjur/cucumber/host/test-app")" > "gcp_token_standard_format"
+                ./get_gce_tokens_to_files.sh || exit 1
                 '''
+                stash name: 'authnGceTokens', includes: 'gcp_token_*', allowEmpty:false
+              }
+            }
+          }
+          post {
+           failure {
+            script {
+              env.GCP_ENV_ERROR = "true"
+            }
+           }
+           success {
+            script {
+              env.GCE_TOKENS_FETCHED = "true"
+            }
+            echo '-- Finished fetching GCE tokens.'
+           }
+          }
+        }
+        /**
+        * We have 3 stages for GCP Authenticator tests. The first one runs inside
+        * In this stage, Google SDK container executes a script to deploy a function,
+        * the function accepts audience in query string and returns a token with that audience.
+        * All the tokens required for testings are obtained and written to function directory, the post stage branch
+        * deletes the function.
+        * This stage depends on stage: 'GCP - Stage A - Compute Engine' to get the GCP project.
+        */
+        stage('GCP - Stage B - Google Function') {
+          environment {
+            GCP_FETCH_TOKEN_FUNCTION = "fetch_token_${BUILD_NUMBER}"
+            IDENTITY_TOKEN_FILE = 'identity-token'
+            GCP_OWNER_SERVICE_KEY_FILE = "sa-key-file.json"
+            GCP_ZONE="us-central1"
+          }
+          steps {
+            echo "Waiting for GCP project name"
```

Comment:
please change to `Waiting for GCP project name (is set by stage 'GCP Authenticator preparation - Allocate GCE Instance')`